src/main/java/org/springframework/pipes/module/launcher/ModuleLauncher.java

```diff
@@ -85,7 +91,19 @@ public void run() {
 				Thread.currentThread().setContextClassLoader(classLoader);
 				new SpringApplicationBuilder(jarFileArchive.getMainClass())
 						.resourceLoader(new DefaultResourceLoader(classLoader))
-						.run("--spring.jmx.default-domain=module-" + new Random().nextInt());
+						.initializers(new ApplicationContextInitializer<ConfigurableApplicationContext>() {
```

Comment:
maybe this could be defined as an inner class rather than anonymous (taking module and environment as ctor args)?